lib/browser.ts

```diff
@@ -837,79 +848,85 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
       return 'Protractor.get(' + destination + ') - ' + str;
     };
 
+    let retPromise: wdpromise.Promise<any>;
+    let then = (fn: () => wdpromise.Promise<any>) => {
```

Comment:
This is clever, but naming it `then` is a little confusing for me, since it makes it hard to distinguish chaining from calling `.then` on a promise at a glance. Maybe just `chainThenables`, `andThen`, or something like that?

Also, why do we need this? Why wouldn't regular promise chaining work?